tests/Feature/ResumeTest.php

```diff
@@ -0,0 +1,217 @@
+
+        // Check an upload for a nonexistent user fails
+        $response = $this->actingAs($user, 'web')->post('/api/v1/users/'.$fakeUserId.'/resume', [
+            'resume' => $pdfOnePage,
+        ]);
+        $response->assertStatus(422);
+        Storage::disk('local')->assertMissing('resumes/'.$fakeUserId.'.pdf');
+
+        // Check an upload for another user fails
+        $response = $this->actingAs($user, 'web')->post('/api/v1/users/'.$alternateId.'/resume', [
+            'resume' => $pdfOnePage,
+        ]);
+        $response->assertStatus(403);
+        Storage::disk('local')->assertMissing('resumes/'.$alternateId.'.pdf');
+
+        // Check an upload with no dues transaction fails
+        $response = $this->actingAs($user, 'web')->post('/api/v1/users/'.$user->id.'/resume', [
+            'resume' => $pdfOnePage,
+        ]);
+        $response->assertStatus(400);
+        Storage::disk('local')->assertMissing('resumes/'.$user->uid.'.pdf');
+        $response->assertJson(static function (AssertableJson $json): void {
+            $json->where('status', 'error')
+                ->where('message', 'inactive');
+        });
+
+        // Create dues package / transaction / payment
+        $package = DuesPackage::factory()->make();
+        $package->effective_end = now()->addMonths(10);
+        $package->save();
+        $transaction = new DuesTransaction();
+        $transaction->user_id = $user->id;
+        $transaction->dues_package_id = $package->id;
+        $transaction->save();
+        $payment = new Payment();
+        $payment->payable_id = $transaction->id;
+        $payment->payable_type = DuesTransaction::getMorphClassStatic();
+        $payment->method = 'cash';
+        $payment->amount = $package->cost;
+        // @phan-suppress-next-line PhanTypeMismatchPropertyProbablyReal
+        $payment->processing_fee = 0;
+        $payment->save();
+
+        // Ideally this would check that uploading no file fails nicely, but there's no easy way to fake that.
+
+        // Check an upload of a PNG fails
+        $response = $this->actingAs($user, 'web')->post('/api/v1/users/'.$user->id.'/resume', [
+            'resume' => $image,
+        ]);
+        $response->assertStatus(400);
+        $response->assertJson(static function (AssertableJson $json): void {
+            $json->where('status', 'error')
+                ->where('message', 'resume_not_pdf');
+        });
+        Storage::disk('local')->assertMissing('resumes/'.$user->uid.'.pdf');
+
+        // Check an upload of a TXT fails
+        $response = $this->actingAs($user, 'web')->post('/api/v1/users/'.$user->id.'/resume', [
+            'resume' => $textFile,
+        ]);
+        $response->assertStatus(400);
+        $response->assertJson(static function (AssertableJson $json): void {
+            // This should be resume_not_pdf, but exiftool does not handle txt files nicely.
```

Comment:
Perhaps we could use the `file` command to guess the type before calling `exiftool`.

Reply:
Good point. I'll open an issue.